codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -160,17 +160,22 @@ public V getAndRemove(K name, V defaultValue) {
     public List<V> getAll(K name) {
         checkNotNull(name, "name");
 
-        LinkedList<V> values = new LinkedList<V>();
-
         int h = hashingStrategy.hashCode(name);
         int i = index(h);
         HeaderEntry<K, V> e = entries[i];
-        while (e != null) {
+
+        if (e == null) {
```

Comment:
Previously, the return value was mutable, so callers could modify their empty copy.   This changes it to be immutable, which may break consumers.

Reply:
+1...